### ISSUE TITLE
epic: Basic Syncer version skew checks and reporting

### ISSUE THREAD

**Issue by davidfestal** [kcp-dev/contrib-tmc#83]
### TL;DR

Ensure that incompatible syncer installations are correctly detected, managed and reported.  

### Progress tracking <a href="#progress-tracking" id="progress-tracking">#</a>

EPIC detailed issues and overall progress can be tracked with the following [project view](https://github.com/orgs/kcp-dev/projects/1/views/49)

### Demo Objective

- [ ] The syncer running should check that its kcp version corresponds to, or is compatible with, the kcp version of the CLI `kcp workload sync` command that the user used to create, or most recently update, the corresponding SyncTarget
- [ ] The Syncer virtual workspace should check that the kcp version of the syncer that accesses it corresponds to, or is compatible with, its own kcp version.
- [ ] Any version discrepancy, along with its criticity (warning or error) should be reported in the SyncTarget, so thatr the end user would be aware of it.
- [ ] Any critical version discrepancy (related with incompatible changes) should prevent the Syncer to start, or syncing of resources to happen.  

### Action Items

- [X] Scope of the current demo is necessary to fit in the prototype boundaries

### Stories

- [ ] kcp-dev/kcp#2351
- [ ] kcp-dev/contrib-tmc#65
- [ ] kcp-dev/contrib-tmc#81
